File: archives/1002.py
import math

T = int(input())
for _ in range(T):
    x1, y1, r1, x2, y2, r2 = map(int, input().split())
    if (x1 == x2) and (y1 == y2):
        if (r1 == r2):
            print(-1)
        else:
            print(0)
    else:
        d1 = math.sqrt((x1-x2)**2 + (y1-y2)**2)
        if d1 < r1 + r2 and d1 > abs(r1 - r2):
            print(2)
        elif d1 == r1 + r2 or d1 == abs(r1 - r2):
            print(1)
        else:
            print(0)

File: archives/cpp/1002.cpp
#include <bits/stdc++.h>
#define FOR(i, start, end) for (int i = start; i < end; i++)
#define endl "\n"
using namespace std;

double distance(int x1, int x2, int y1, int y2) {
    return sqrt((x2-x1) * (x2-x1) + (y2-y1) * (y2-y1));
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int T; cin >> T;
    while (T--) {
        int x1, y1, r1, x2, y2, r2;
        cin >> x1 >> y1 >> r1 >> x2 >> y2 >> r2;

        if ((x1 == x2) && (y1 == y2)) {
            if (r1  == r2) {
                cout << -1 << endl;
            } else {
                cout << 0 << endl;
            }
        } else {
            double dist = distance(x1, x2, y1, y2);
            if (dist < r1+r2 && dist > abs(r1-r2)) {
                cout << 2 << endl;
            } else if (dist == (double)(r1+r2) || dist == (double)abs(r1-r2)) {
                cout << 1 << endl;
            } else {
                cout << 0 << endl;
            }
        }
    }
    return 0;
}